### PR TITLE
(PUP-7864) Add Puppet Type Loader for Tasks

### PR DESCRIPTION
**Pull request by thallgren**
This commit adds a loader that finds files under the 'tasks' folder
in a module root directory. The loader will expect to find exactly one
file that matches the expected task name and has an optional or
arbitrary file extension that is different from '.json'. That file may
optionally be accompanied by file that has the same name and a '.json'
extension. The json latter is then the metadata that describes the task
type.

### CODE REVIEW

File: lib/puppet/pops/types/task.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+require 'json'
+
+module Puppet::Pops
+module Types
+  # Pcore variant of the Adaptable::Adapter. Uses a Pcore Object type instead of a Class
```
Comment (hlindberg) [inline lib/puppet/pops/types/task.rb]: This comment looks like copy pasta...

File: lib/puppet/pops/types/task.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  class Task
+    include PuppetObject
+
+    # Register the Annotation type. This is the type that all custom Annotations will inherit from.
```
Comment (hlindberg) [inline lib/puppet/pops/types/task.rb]: This comment looks like copy pasta...

File: spec/unit/pops/types/task_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+require 'spec_helper'
+require 'puppet/pops'
+require 'puppet_spec/compiler'
+
+module Puppet::Pops
+module Types
+describe 'The Tast Type' do
```
Comment (hlindberg) [inline spec/unit/pops/types/task_spec.rb]: typo "Tast"

File: lib/puppet/pops/loader/task_instantiator.rb [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    name = typed_name.name
+    metadata = nil
+    task_source = nil
+    source_refs.each do |source_ref|
+      if source_ref.end_with?('.json')
+        metadata = source_ref
+      elsif task_source.nil?
+        task_source = source_ref
+      else
+        raise ArgumentError, _('Only one file can exists besides the .json file for task %{name} in directory %{directory}') %
```
Comment (MosesMendoza) [inline lib/puppet/pops/loader/task_instantiator.rb]: I think we want s/exists/exist

**Pull request merged**